modules/core/test/test_mat.cpp

```diff
@@ -1248,6 +1248,29 @@ static void check_ndim_shape(const cv::Mat &mat, int cn, int ndims, const int *s
         EXPECT_EQ(mat.size[i], sizes[i]);
 }
 
+TEST(Core_Mat, convertTo)
+{
+    cv::Mat A(8, 16, CV_8UC3, cv::Scalar(1, 2, 3));
+    void* p = A.data;
+    int cn = CV_MAT_CN(CV_8UC3);
+    
+    {
+        int depth = CV_8S;
+        int dtype = CV_MAKE_TYPE(depth, cn);
+        A.convertTo(A, depth);
+        EXPECT_EQ(p, A.data);
+        EXPECT_EQ(A.type(), dtype);
+    }
+
+    {
+        int depth = CV_32F;
+        int dtype = CV_MAKE_TYPE(depth, cn);
+        A.convertTo(A, depth);
+        EXPECT_NE(p, A.data);
+        EXPECT_EQ(A.type(), dtype);
+    }
+}
```

Comment:
Please also check the following case with existing reference
```cpp
cv::Mat A(8, 16, CV_8UC3, cv::Scalar(1, 2, 3));
cv::Mat B = A;
A.convertTo(A, CV_8SC3);
```

Reply:
I have got your idea, in this case, A still need to allocate new memory.

Reply:
Agree. So let's add a test case too.

Reply:
OK,  I added a check for the reference count in the convertTo function.

Reply:
I think just better to highlight final types and data ptrs:
```
A.type == new_type
B.type == old_type
A.ptr != B.ptr
```

Reply:
Okay, I've already made the modifications to the test case and the convertTo code. Thanks you very much.